src/stream-tcp-reassemble.c

```diff
@@ -728,10 +728,14 @@ int StreamTcpReassembleHandleSegmentHandleData(ThreadVars *tv, TcpReassemblyThre
 
     /* If we have reached the defined depth for either of the stream, then stop
        reassembling the TCP session */
+    bool depth_reached = (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED);
     uint32_t size = StreamTcpReassembleCheckDepth(ssn, stream, TCP_GET_SEQ(p), p->payload_len);
     SCLogDebug("ssn %p: check depth returned %"PRIu32, ssn, size);
 
     if (stream->flags & STREAMTCP_STREAM_FLAG_DEPTH_REACHED) {
+        if (!depth_reached)
```

Comment:
I don't understand this, if we get here multiple times the counter would also be incremented more than once per stream?

Reply:
The thinking is that one event per stream instead of once per occurrence.

What's your preference?

Reply:
The logic is supposed to be once per stream both for the event and the counter increment.

Reply:
Will adjust to only count and raise the event the first time the limit is reached.

Reply:
It *should* already work that way. If not, there is a bug in the existing code.